### ISSUE TITLE
Too many clauses send to elastic search on opening newspaper migration page

### ISSUE THREAD

**Issue by henning-gerhardt** [kitodo/kitodo-production#4027]
While trying to load page to migrate newspapers the following error get logged:

```
07-Oct-2020 11:41:41.897 SEVERE [http-nio-127.0.0.1-8080-exec-7] org.omnifaces.exceptionhandler.FullAjaxExceptionHandler.logException FullAjaxExceptionHandler: An exception occurred
during rendering JSF ajax response. Error page '/pages/error.jsf' will be shown.
        ElasticsearchStatusException[Elasticsearch exception [type=search_phase_execution_exception, reason=all shards failed]]
                at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:177)
                at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:526)
                at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:502)
                at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:409)
                at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:382)
                at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:323)
                at org.kitodo.data.elasticsearch.search.SearchRestClient.getDocument(SearchRestClient.java:185)
                at org.kitodo.data.elasticsearch.search.Searcher.findDocuments(Searcher.java:204)
                at org.kitodo.data.elasticsearch.search.Searcher.findDocuments(Searcher.java:163)
                at org.kitodo.production.services.data.base.SearchService.findByQuery(SearchService.java:449)
                at org.kitodo.production.services.data.base.ClientSearchService.findByQuery(ClientSearchService.java:128)
                at org.kitodo.production.services.data.base.ProjectSearchService.findByQuery(ProjectSearchService.java:77)
                at org.kitodo.production.services.data.base.SearchService.convertRelatedJSONObjectToDTO(SearchService.java:543)
                at org.kitodo.production.services.data.BatchService.convertRelatedJSONObjects(BatchService.java:165)
                at org.kitodo.production.services.data.BatchService.convertJSONObjectToDTO(BatchService.java:159)
                at org.kitodo.production.services.data.BatchService.convertJSONObjectToDTO(BatchService.java:41)
                at org.kitodo.production.services.data.base.SearchService.convertJSONObjectsToDTOs(SearchService.java:516)
                at org.kitodo.production.services.data.base.SearchService.findAll(SearchService.java:140)
                at org.kitodo.production.migration.NewspaperProcessesMigrator.getNewspaperBatches(NewspaperProcessesMigrator.java:221)
                at org.kitodo.production.forms.MigrationForm.getNewspaperBatchesItems(MigrationForm.java:501)
                at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.lang.reflect.Method.invoke(Method.java:498)
                at javax.el.BeanELResolver.getValue(BeanELResolver.java:94)
                at javax.el.CompositeELResolver.getValue(CompositeELResolver.java:62)

... shorted ...

               at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
               at java.lang.Thread.run(Thread.java:748)
                Suppressed: org.elasticsearch.client.ResponseException: POST http://sdvkitodoes20.slub-dresden.de:9200/kitodo/process/_search?typed_keys=true&ignore_unavailable=false
&expand_wildcards=open&allow_no_indices=true&search_type=query_then_fetch&batched_reduce_size=512: HTTP/1.1 400 Bad Request
{"error":{"root_cause":[{"type":"query_shard_exception","reason":"failed to create query: {\n  \"bool\" :

... shorted as a few thousand entries (looks like Kitodo IDs) are following

n    \"disable_coord\" : false,\n    \"adjust_pure_negative\" : true,\n    \"boost\" : 1.0\n  }\n}","index_uuid":"N_uTGldrSO2RNpo_E4ruLA","index":"kitodo","caused_by":{"typ
e":"too_many_clauses","reason":"maxClauseCount is set to 1024"}}}]},"status":400}
                        at org.elasticsearch.client.RestClient$1.completed(RestClient.java:354)
                        at org.elasticsearch.client.RestClient$1.completed(RestClient.java:343)
                        at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:119)
                        at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:177)
                        at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:436)
                        at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:326)
                        at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
                        at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
                        at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
                        at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
                        at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
                        at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
                        at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
                        at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
                        at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
                        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:588)
                        ... 1 more
07-Oct-2020 11:42:01.885 SEVERE [http-nio-127.0.0.1-8080-exec-2] org.apache.myfaces.context.servlet.ServletExternalContextImpl.setResponseContentType Cannot set content type. Respons
e already committed
```

If I read the error log correct, main reason is, that there too many clauses (Kitodo Ids) are send to elastic search and only 1024 are allowed. We have newspaper batches with more than 40.000 entries.

used version: master branch 2020-10-05 / commit 6c5b92a674159520c1929549b80e44b5568be99f

**Comment by matthias-ronge** [comment]
This is not due to the migration service, but to the batch service. [When the BatchDTO is built, all processes are fetched from the index using the ID.](../../kitodo-production/blob/master/Kitodo/src/main/java/org/kitodo/production/services/data/BatchService.java#L165) Unlike Hibernate, there is no lazy loading or the like. It is probably sufficient to increase the limit for query components in ElasticSearch. It basically makes sense that there are such limits, otherwise you could easily run a denial-of-service attack (e.g. with a search engine that is integrated into a website). In our case, however, I don't see any danger if you raise this limit.

Basically, maybe we could reprogram the migration service so that it uses the database directly. But I'm afraid it'll be even slower than it already is, something I'd like to avoid.
